dali/operators/reader/gds_mem_test.cu

```diff
@@ -35,52 +35,72 @@ struct CUFileDriverScope {
   }
 };
 
-TEST(GDSMem, AllocatorMultiDevice) {
-  CUFileDriverScope scope;
-  int ndev;
-  CUDA_CALL(cudaGetDeviceCount(&ndev));
-  if (ndev < 2) {
-    GTEST_SKIP() << "This test requires more than one CUDA capable device to run.";
-    return;
+template <typename TestBody>
+void SkipIfIncompatible(TestBody &&body) {
+  try {
+    body();
+  } catch (const CUFileError &e) {
+    if (e.result().err == CU_FILE_PLATFORM_NOT_SUPPORTED ||
+        e.result().err == CU_FILE_DEVICE_NOT_SUPPORTED) {
+      GTEST_SKIP() << "No GDS-capable device found";
+    } else {
+      throw;
+    }
   }
-  ASSERT_NE(GDSAllocator::get(0), GDSAllocator::get(1));
+}
+
+TEST(GDSMem, AllocatorMultiDevice) {
+  SkipIfIncompatible([&]{
```

Comment:
Starting from here, the only change is wrapping everything in SkipIfIncompatible - the rest is just indentation.